### PR TITLE
Fixed issues with udChunkedArray iterators

### PR DESCRIPTION
**Pull request by samsinsane**
- Added unit test to ensure that jumping backwards on an iterator works

### CODE REVIEW

File: Include/udChunkedArray.h [hunk 1]
```diff
@@ -53,6 +53,16 @@ struct udChunkedArrayIterator
...
 
+template <typename T>
+inline udChunkedArrayIterator<T> operator+(typename udChunkedArrayIterator<T>::difference_type a, udChunkedArrayIterator<T> b) {
+  b += a; return b;
+}
+
+template <typename T>
+inline udChunkedArrayIterator<T> operator-(typename udChunkedArrayIterator<T>::difference_type a, udChunkedArrayIterator<T> b) {
+  b += a; return b;
+}
```
Comment (OsiJr) [inline Include/udChunkedArray.h:64]: Is this to be consistent with a different coding standard?

File: Include/udChunkedArray.h [hunk 2]
```diff
@@ -92,6 +102,16 @@ struct udChunkedArrayConstIterator
...
 };
 
+template <typename T>
+inline udChunkedArrayConstIterator<T> operator+(typename udChunkedArrayConstIterator<T>::difference_type a, udChunkedArrayConstIterator<T> b) {
+  b += a; return b;
+}
+
+template <typename T>
+inline udChunkedArrayConstIterator<T> operator-(typename udChunkedArrayConstIterator<T>::difference_type a, udChunkedArrayConstIterator<T> b) {
+  b += a; return b;
```
Comment (OsiJr) [inline Include/udChunkedArray.h:112]: Just want a confirm that `a` is negative here (adding the negative would then be correct).

File: Include/udChunkedArray.h (hunk 1 above)
Comment (samsinsane) [reply Include/udChunkedArray.h:64]: Ah, I'm not sure what it's on one line. I'll fix that up.

File: Include/udChunkedArray.h (hunk 2 above)
Comment (samsinsane) [reply Include/udChunkedArray.h:112]: I don't think `a` is negative, I think I just goofed it. I only added a unit test to trigger the edge case of moving the iterator backwards, which doesn't use this particular operator. I'll add this operator into the normal unit tests.

**Review by OsiJr — APPROVED** [review]
